### PR TITLE
Restore scifi clusters mc

### PR DESCRIPTION
**Pull request by siilieva**
- bring back Scifi cluster making for the run_digi task
- restructure the reverted commits (includes comment cleaning)
- add option to run digi without making the clusters (by default clusters will be created)

! Sorting out the cluster-making issue #198 is beyond the intended scope of this PR.

### CODE REVIEW

File: shipLHC/run_digiSND.py [hunk 1]
```diff
@@ -28,10 +28,13 @@ def mem_monitor():
 parser.add_argument("-ss", "--saturationScifi", dest="ss", type=float, help="saturation energy for Scifi [p.e.]", default=104.)
 parser.add_argument("-tML", "--thresholdMufiL", dest="tml", type=float, help="threshold energy for Mufi large [p.e.]", default=0.0)
 parser.add_argument("-tMS", "--thresholdMufiS", dest="tms", type=float, help="threshold energy for Mufi small [p.e.]", default=0.0)
-parser.add_argument("-cpp", "--digiCPP", action='store_true', dest="FairTask_digi", help="perform digitization using DigiTaskSND", default=False)
+parser.add_argument("-no-cls", "--noClusterScifi", action='store_true', dest="noClusterScifi", help="do not make Scifi clusters")
```
Comment (olantwin) [inline shipLHC/run_digiSND.py:31]: I would remove the short option here. Multi-character short options are quite unconventional.
For the long option, could we do kebab-case please? ;)
If we stick to `--noClusterScifi`, the `dest=` isn't needed, as it's automatically `noClusterScifi`.

File: sndFairTasks/DigiTaskSND.cxx [hunk 2]
```diff
@@ -103,7 +107,7 @@ void DigiTaskSND::Exec(Option_t* /*opt*/)
 {
 
     fScifiDigiHitArray->Delete();
-    fScifiClusterArray->Delete();
+    if (fMakeClusterScifi) fScifiClusterArray->Delete();
```
Comment (olantwin) [inline sndFairTasks/DigiTaskSND.cxx:110]: This should be `Clear()` or `Clear('C')`, see Note 2 [here](https://root.cern/doc/master/classTClonesArray.html)

File: python/SndlhcDigi.py [hunk 3]
```diff
@@ -57,9 +60,10 @@ def digitize(self):
         self.digitizeScifi()
         self.digiScifiBranch.Fill()
         self.digiScifi2MCPointsBranch.Fill()
-        self.clusScifi.Delete()
-        self.clusterScifi()
-        self.clusScifiBranch.Fill()
+        if self.makeClusterScifi:
+           self.clusScifi.Delete()
```
Comment (olantwin) [inline python/SndlhcDigi.py:64]: This should be `Clear()` or `Clear('C')`, see Note 2 [here](https://root.cern/doc/master/classTClonesArray.html)

**Review by olantwin — CHANGES_REQUESTED** [review]
* I don't understand why ff0d56b1fc71ec27ff9f4d37407252f6c869c98e was reverted, us you redo the changes of the reverted commit in 7a1b236f2ebcfccb29d7fc2dd9e8032877af80de.
* f8176d24554d8222ce1720fe6ea253ba7a9687e1 should be a fixup to e598f31e281848ecd4d251163ea51117f4cfedbf.

LGTM otherwise. Happy to defer general improvements to a future PR addressing #198 .

**Comment by siilieva** [comment]
I restructured the commits for this PR as the task to remove the thresholds was unnecessarily split in two commits before.
Since restructuring was already hapening for the reason above, I made a new commit for the clear up.

File: shipLHC/run_digiSND.py (hunk 1 above)
Comment (siilieva) [reply shipLHC/run_digiSND.py:31]: I find the` no-<smth>` flags very clear, even if longer than optimal for a short option. Still, it is sure shorter than the long one here `noClusterScifi`:D
Here I am simply following the case style of the other options.
Ok for _dest_, will edit the commit shortly.

File: python/SndlhcDigi.py (hunk 3 above)
Comment (siilieva) [reply python/SndlhcDigi.py:64]: same as above :+1:

File: shipLHC/run_digiSND.py (hunk 1 above)
Comment (olantwin) [reply shipLHC/run_digiSND.py:31]: Usually the convention is that the single-dash options are a single character. I'd rather have no short option in cases where this is not possible. We don't need short options for everything.

File: sndFairTasks/DigiTaskSND.cxx (hunk 2 above)
Comment (olantwin) [reply sndFairTasks/DigiTaskSND.cxx]: Great! I've actually wanted to make this change for a while, but never got around to it. Should reduce the time and memory needed for digitisation.

**Comment by olantwin** [comment]
> I restructured the commits for this PR as the task to remove the thresholds was unnecessarily split in two commits before. Since restructuring was already hapening for the reason above, I made a new commit for the clear up.

Unless you want the reverts to remain exactly as is, I would still suggest just making f8176d24554d8222ce1720fe6ea253ba7a9687e1 and 7a1b236f2ebcfccb29d7fc2dd9e8032877af80de fixups to the respective reverts to clean up the history a bit.

Otherwise things look OK now.

**Comment by siilieva** [comment]
> > I restructured the commits for this PR as the task to remove the thresholds was unnecessarily split in two commits before. Since restructuring was already hapening for the reason above, I made a new commit for the clear up.
> 
> Unless you want the reverts to remain exactly as is, I would still suggest just making [f8176d2](https://github.com/SND-LHC/sndsw/commit/f8176d24554d8222ce1720fe6ea253ba7a9687e1) and [7a1b236](https://github.com/SND-LHC/sndsw/commit/7a1b236f2ebcfccb29d7fc2dd9e8032877af80de) fixups to the respective reverts to clean up the history a bit.
> 
> Otherwise things look OK now.

I tried fixup-ing locally and it looks weird. I prefer the way we have currently with separate revert and modifications commits.

**Review by olantwin — APPROVED** [review]

**Comment by dcentanni** [comment]
Dear Simona and Oliver, it looks OK to me. Thank you!

**Pull request merged**